cmd/list.go

```diff
@@ -0,0 +1,44 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/codegangsta/cli"
+	"github.com/exercism/cli/api"
+	"github.com/exercism/cli/config"
+)
+
+const msgExplainFetch = "In order to fetch a specific assignment, call the fetch command with a specific assignment.\n\nexercism fetch ruby/matrix"
+
+// List returns the full list of assignments for a given language
+func List(ctx *cli.Context) {
+	c, err := config.Read(ctx.GlobalString("config"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	args := ctx.Args()
+
+	if len(args) != 1 {
+		msg := "Usage: execism list LANGUAGE"
+		log.Fatal(msg)
+	}
+
+	language := args[0]
+
+	problems, err := api.List(language, c.XAPI)
+	if err != nil {
+		log.Fatal(err)
+	}
+	GenerateOutput(problems, os.Stdout)
+}
+
+// Output writes out the list of assignments
+func GenerateOutput(problems []string, w io.Writer) {
```

Comment:
I think I was having a ruby moment here. Of course `GenerateOutput` is going to be available to everything in the `cmd` package. The motivation behind adding the function was to allow me to easily test the generated output. However, the test is quite brittle and adds little value.

My leaning would be towards:
* changing the function name to be something unexported and more appropriate like: `outputProblemList` -or-
* folding the function in to `cmd.List` and only abstracting as required

I'd favour eliminating the function for now. What do you think about doing this for now and exploring the value of a `track` package once we merge this in ?